### ISSUE TITLE
Message stuck in queue after failing deliver to mda

### ISSUE THREAD

**Issue by breard-r** [OpenSMTPD/OpenSMTPD#855]
Hi,

I have a message stuck in queue after OpenSMTPD failed to deliver it to the mda.
I am running OpenSMTPD 6.0.3 portable on ArchLinux.
The important part of my configuration file is (was) the following:

`accept for local alias <aliases> deliver to mda "/usr/lib/dovecot/dovecot-lda -e -f %{sender} -d %{user.username}"`

## What happened?

I sent an email to someone who as a 451 "Temporary local problem" error. After 4 hours in the queue, OpenSMTPD sent me an email notification so I would know about it. When this notification was delivered to me, a first problem appeared: the `%{sender}` was set to `@` instead of something like `MAILER-DAEMON@domain.example.com`. It causes dovecot-lda to fail with "lda: Fatal: Invalid -f parameter: Missing domain after '@' in source route".

I removed the `-f %{sender}` from my configuration and then tried to get the message delivered by:
1. restarting OpenCMTPD
2. `smtpctl pause envelope_id` then `smtpctl resume envelope_id`
3. `smtpctl schedule envelope_id`

And here is the second problem: none of these commands worked. When I run `smtpctl show envelope envelope_id` I can see the following:

```
errorline: "lda: Fatal: Invalid -f parameter: Missing domain after '@' in source route"
mda-buffer: /usr/lib/dovecot/dovecot-lda -e -f @ -d rodolphe
```

It seems like the command to deliver a message is cached and can never be changed, wich result in message being stuck in the queue if the command was invalid.

## Summary

- For automatic emails delivery, `%{sender}` is not correctly set.
- When the MDA delivery command is invalid, it is impossible to change it and the message is stuck.

**Comment by poolpOrg** [comment]
This ticket should be split in two different tickets:

- one for the '@' sender as this needs a fix
- one for the 'envelopes' stuck in queue with the broken command, as this fixed in new grammar

**Comment by poolpOrg** [comment]
actually the part that isn't fixed is duplicate from #793, it is the exact same cause, closing this ticket
